src/Components/Shifting/BadgesList.tsx

```diff
@@ -1,80 +1,41 @@
-import { getAnyFacility, getUserList } from "../../Redux/actions";
-import { useEffect, useState } from "react";
-
 import { SHIFTING_FILTER_ORDER } from "../../Common/constants";
-import { useDispatch } from "react-redux";
+
 import { useTranslation } from "react-i18next";
+import useQuery from "../../Utils/request/useQuery";
+import routes from "../../Redux/api";
+
+function useFacilityQuery(facilityId: string | undefined) {
+  return useQuery(routes.getAnyFacility, {
+    pathParams: { id: String(facilityId) },
+    prefetch: facilityId !== undefined,
+  });
+}
 
 export default function BadgesList(props: any) {
   const { qParams, FilterBadges } = props;
-  const [assignedUsername, setAssignedUsername] = useState("");
-  const [assignedFacilityName, setAssignedFacilityName] = useState("");
-  const [originFacilityName, setOriginFacilityName] = useState("");
-  const [approvingFacilityName, setApprovingFacilityName] = useState("");
-  const dispatch: any = useDispatch();
+
   const { t } = useTranslation();
 
   const booleanFilterOptions = {
     trueValue: t("yes"),
     falseValue: t("no"),
   };
 
-  useEffect(() => {
-    async function fetchData() {
-      if (!qParams.assigned_to) return setAssignedUsername("");
-      const res = await dispatch(
-        getUserList({ id: qParams.assigned_to }, "assigned_user_name")
-      );
-      const { first_name, last_name } = res?.data?.results[0] || {};
-      setAssignedUsername(`${first_name} ${last_name}`);
-    }
-    fetchData();
-  }, [dispatch, qParams.assigned_to]);
-
-  useEffect(() => {
-    async function fetchData() {
-      if (!qParams.origin_facility) return setOriginFacilityName("");
-      const res = await dispatch(
-        getAnyFacility(qParams.origin_facility, "origin_facility_name")
-      );
-      setOriginFacilityName(res?.data?.name);
-    }
-    fetchData();
-  }, [dispatch, qParams.origin_facility]);
-
-  useEffect(() => {
-    async function fetchData() {
-      if (!qParams.shifting_approving_facility)
-        return setApprovingFacilityName("");
-      const res = await dispatch(
-        getAnyFacility(
-          qParams.shifting_approving_facility,
-          "shifting_approving_facility_name"
-        )
-      );
-      setApprovingFacilityName(res?.data?.name);
-    }
-    fetchData();
-  }, [dispatch, qParams.shifting_approving_facility]);
+  const { data: assignedUser } = useQuery(routes.userList, {
+    query: { id: qParams.assigned_to },
+    prefetch: qParams.assigned_to ? true : false,
```

Comment:
Why do: `if (true) return true else false` 😅

```suggestion
    prefetch: !!qParams.assigned_to,
```